pure+static cgo does not link symbols from "runtime/cgo"

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of rules_go are you using?

0.19.4 and HEAD (c44823f9d83597fd22cfed75f063aefc5de429eb).

### What version of gazelle are you using?

N/A

### What version of Bazel are you using?

0.28.0

### Does this issue reproduce with the latest releases of all the above?

Yes.

### What operating system and processor architecture are you using?

Linux, x86_64

### Any other potentially useful information about your toolchain?

Issue present with both gcc and llvm with lld linker.

### What did you do?

```
bazel build --features=static //tests/core/cgo:pure
```

### What did you expect to see?

A successful build.

### What did you see instead?

```
external/go_sdk/pkg/tool/linux_amd64/link: running /usr/bin/gcc failed: exit status 1
/tmp/go-link-689103257/go.o:go.go:_cgo_callers: error: undefined reference to 'x_cgo_callers'
/tmp/go-link-689103257/go.o:go.go:_cgo_init: error: undefined reference to 'x_cgo_init'
/tmp/go-link-689103257/go.o:go.go:_cgo_mmap: error: undefined reference to 'x_cgo_mmap'
/tmp/go-link-689103257/go.o:go.go:_cgo_munmap: error: undefined reference to 'x_cgo_munmap'
/tmp/go-link-689103257/go.o:go.go:_cgo_notify_runtime_init_done: error: undefined reference to 'x_cgo_notify_runtime_init_done'
/tmp/go-link-689103257/go.o:go.go:_cgo_sigaction: error: undefined reference to 'x_cgo_sigaction'
/tmp/go-link-689103257/go.o:go.go:_cgo_thread_start: error: undefined reference to 'x_cgo_thread_start'
/tmp/go-link-689103257/go.o:go.go:runtime._cgo_setenv: error: undefined reference to 'x_cgo_setenv'
/tmp/go-link-689103257/go.o:go.go:runtime._cgo_unsetenv: error: undefined reference to 'x_cgo_unsetenv'
/tmp/go-link-689103257/go.o:go.go:runtime.cgo_yield: error: undefined reference to '_cgo_yield'
```

--- a/go/private/actions/link.bzl
+++ b/go/private/actions/link.bzl
@@ -79,7 +79,7 @@ def emit_link(
         tool_args.add("-race")
     if go.mode.msan:
         tool_args.add("-msan")
-    if go.mode.static or go.mode.link != LINKMODE_NORMAL:
+    if (go.mode.static and not go.mode.pure) or go.mode.link != LINKMODE_NORMAL:
         tool_args.add("-linkmode", "external")
     if go.mode.static:
         extldflags.append("-static")

--- a/tests/core/go_binary/BUILD.bazel
+++ b/tests/core/go_binary/BUILD.bazel
@@ -109,6 +109,7 @@ go_test(
     srcs = ["static_test.go"],
     data = select({
         "@io_bazel_rules_go//go/platform:linux": [
+            ":static_bin",
             ":static_cgo_bin",
             ":static_pure_bin",
         ],
@@ -116,6 +117,14 @@ go_test(
     }),
     rundir = ".",
     deps = ["//go/tools/bazel:go_default_library"],
+)
+
+go_binary(
+    name = "static_bin",
+    srcs = ["static_bin.go"],
+    deps = ["@org_golang_x_sys//unix:go_default_library"],
+    static = "on",
+    tags = ["manual"],
 )
 
 go_binary(
@@ -129,6 +138,7 @@ go_binary(
 go_binary(
     name = "static_pure_bin",
     srcs = ["static_pure_bin.go"],
+    pure = "on",
     static = "on",
     tags = ["manual"],
 )

--- a/tests/core/go_binary/static_bin.go
+++ b/tests/core/go_binary/static_bin.go
@@ -1,0 +1,7 @@
+package main
+
+import (
+	_ "golang.org/x/sys/unix" // See https://github.com/bazelbuild/rules_go/issues/2168
+)
+
+func main() {}

--- a/tests/core/go_binary/static_test.go
+++ b/tests/core/go_binary/static_test.go
@@ -24,7 +24,7 @@ import (
 )
 
 func TestStatic(t *testing.T) {
-	for _, name := range []string{"static_cgo_bin", "static_pure_bin"} {
+	for _, name := range []string{"static_bin", "static_cgo_bin", "static_pure_bin"} {
 		t.Run(name, func(t *testing.T) {
 			path, ok := bazel.FindBinary("tests/core/go_binary", name)
 			if !ok {

